src/app/entry/archive/dialog/archive-entry-dialog.component.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ *     Copyright 2023 OICR, UCSC
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License")
+ *     you may not use this file except in compliance with the License
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ */
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { Router } from '@angular/router';
+import { Entry, EntriesService, DockstoreTool, Workflow } from '../../../shared/openapi';
+import { AlertService } from '../../../shared/alert/state/alert.service';
+
+@Component({
+  selector: 'app-archive-entry-dialog',
+  templateUrl: './archive-entry-dialog.component.html',
+  styleUrls: ['./archive-entry-dialog.component.scss'],
+})
+export class ArchiveEntryDialogComponent {
+  clicked: boolean;
+  term: string;
+  path: string;
+  fromDockstoreYml: boolean;
+
+  constructor(
+    public dialogRef: MatDialogRef<ArchiveEntryDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: { entry: Entry; callback: (archived: Entry) => void },
```

Comment:
Despite our attempts with the OpenAPI hierarchy, DockstoreTool and Workflow unfortunately do not extend Entry. If they did extend `Entry`, this would be the right way.

Since they don't, I'd suggest making this `entry: Workflow | DockstoreTool`. 

Rationale: if we ever introduce a new Entry that doesn't extend DockstoreTool or Workflow, unlikely though that may be, we would get a compilation error, which we would want, because line 46 would fail at runtime.

That said, you have to go through hoops to enforce type checking with MAT_DIALOG_DATA, which we don't do, so I guess optional.

Reply:
I get your drift, the two approaches have different pros and cons, for sure.  Yes, `Workflow` and `DockstoreTool` don't formally derive to `Entry` right now, but we know they are derivatives.  The choice to use `Entry` was intentional on my part, so it's best IMHO, but you're the lead, so feel free to override... :)

As a compromise, I added some code to line 46 that uses the `Entry.gitUrl` as a fallback, so for a radically-different type of new entry, it's more likely to display something moderately useful, at least...